### PR TITLE
Fix linter findings

### PR DESCRIPTION
**Pull request by sofkaski**
~~This PR is stacked on  top of #53~~

This PR has changes:

1. Move linter configuration to pyproject.toml out of `Makefile`. Also disabled the linter rule `format` in favor of `black`
2. Changes to address linter findings. They are split in 3 commits:
   1.  Miscellaneous linter finding fixes.
   2. Disable the linter rule `invalid-overridden-method` in async variant. See the commit message for rationale and possible further development.
   3. Changed logging to use the recommended lazy logging message formatting instead of f-string formatting. (note: f-string formatting could be allowed (by disabling a linter rule))

Also formatted files with `make black`.

Now both `make lint` and `make black` should produce clean results.

### CODE REVIEW

File: pysolarmanv5/pysolarmanv5.py [hunk 1]
```diff
@@ -284,81 +278,89 @@ def _send_receive_v5_frame(self, data_logging_stick_frame):
...
         return v5_response
 
+    def _received_frame_is_valid(self, frame):
+        """Check that the frame is valid and that the serial number of the received
+        frame matches with the last sent one.
+        Ignore also any counter frames.
+        """
+        if frame.startswith(b"\xa5\x01\x00\x10G"):
+            # Frame with control code 0x4710 - Counter frame
+            self.log.debug("[{self.serial}] COUNTER: %s", frame.hex(" "))
```
Comment (githubDante) [inline pysolarmanv5/pysolarmanv5.py:295]: `{self.serial}` should be problematic in this block

File: pysolarmanv5/pysolarmanv5.py [hunk 2]
```diff
@@ -284,81 +278,89 @@ def _send_receive_v5_frame(self, data_logging_stick_frame):
...
+            return False
+        if not frame.startswith(b"\xa5"):
```
Comment (jmccrohan) [inline pysolarmanv5/pysolarmanv5.py:297]: As comment above. Replace with self.v5_start?

File: pysolarmanv5/pysolarmanv5_async.py [hunk 3]
```diff
@@ -1,11 +1,20 @@
...
 from umodbus.client.serial import rtu
 from .pysolarmanv5 import NoSocketAvailableError, PySolarmanV5
 
+# Disable `invalid-overridden-method` rule. The class `PySolarmanV5Async` overrides
+# (=redefines) non-async methods from `PySolarmanV5`.
+# The override in this case is not a problem in practice.
+
+# This could be avoided by changing the class hierarchy.
+# One way would be to introduce a common base class for async and sync classes
+# that would capture attributes and common (internal) sync methods.
```
Comment (jmccrohan) [inline pysolarmanv5/pysolarmanv5_async.py:14]: I'm open to discussing this in a future PR. :slightly_smiling_face:

**Review by jmccrohan — COMMENTED** [review]
Thanks @sofkaski for the work on this PR. Some minor comments added.

File: pysolarmanv5/pysolarmanv5.py (hunk 1 above)
Comment (sofkaski) [reply pysolarmanv5/pysolarmanv5.py:295]: Indeed... will fix

File: pysolarmanv5/pysolarmanv5.py [hunk 4]
```diff
@@ -284,81 +278,89 @@ def _send_receive_v5_frame(self, data_logging_stick_frame):
...
+            self.log.debug("[{self.serial}] V5_MISMATCH: %s", frame.hex(" "))
+            return False
+        if frame[5] != self.sequence_number:
+            self.log.debug("[{self.serial}] V5_SEQ_NO_MISMATCH: %s", frame.hex(" "))
+            return False
```
[the remark this answers is not in the document]
Comment (sofkaski) [reply pysolarmanv5/pysolarmanv5.py]: Fixed.

File: pysolarmanv5/pysolarmanv5.py [hunk 5]
```diff
@@ -284,81 +278,89 @@ def _send_receive_v5_frame(self, data_logging_stick_frame):
...
-        self.log.debug("RECD: " + v5_response.hex(" "))
+        self.log.debug("RECD: %s", v5_response.hex(" "))
         return v5_response
 
+    def _received_frame_is_valid(self, frame):
+        """Check that the frame is valid and that the serial number of the received
+        frame matches with the last sent one.
+        Ignore also any counter frames.
+        """
+        if frame.startswith(b"\xa5\x01\x00\x10G"):
```
[the remark this answers is not in the document]
Comment (sofkaski) [reply pysolarmanv5/pysolarmanv5.py]: I added a comment to the function docstring. For the future: maybe the structure of the message on the byte level could be declared somehow globally and used also in encode/decode.

File: pysolarmanv5/pysolarmanv5.py (hunk 2 above)
Comment (sofkaski) [reply pysolarmanv5/pysolarmanv5.py]: Fixed

File: pysolarmanv5/pysolarmanv5_async.py (hunk 3 above)
Comment (sofkaski) [reply pysolarmanv5/pysolarmanv5_async.py:14]: Let's see. Next I need to look into my battery management integration ...

**Review by githubDante — APPROVED** [review]
It looks good to me.

**Comment by jmccrohan** [comment]
Merged. Thanks @sofkaski!

Thanks @githubDante for the review.

**Pull request merged**